### PR TITLE
Issue 23948 make preview button look pritty

### PR DESCRIPTION
**Pull request by freddyucv**
### Proposed Changes
* Turn the dot-api-link into a new more raw Component: dot-link

This new Component is goin to be use wherever we need a link.

https://github.com/dotCMS/core/pull/24807/files#diff-094d5fd63dfbbf4d1b6012684da83c2c2ec9cd2649d57d8a178d1e456257d8faR1

* Use this new Component inside the dot-api-link

https://github.com/dotCMS/core/pull/24807/files#diff-4097c2aeff1921944cf40f35efa6dd183958902f3a37e9781c975c7686d977a3R1

* Sue the new Component to the Preview Button in the Edit Page Portlet

https://github.com/dotCMS/core/pull/24807/files#diff-a6dc3768a22138335dcd4b74a4517c459e4b5d8ab41dca8e794cebb828751baaR10

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.component.scss [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// This styles are the same as .p-button.p-button-text in _buttons.scss,
+// the problem is the :enabled pseudoclass, hyperlink tag doesn't have that pseudoclass
+
+.p-button {
+    &:hover {
+        background: $color-palette-primary-100;
+    }
+
+    &:focus {
+        background: white;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.component.scss:18]:
```suggestion
        background: $white;
```

File: core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.component.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        box-shadow: $md-shadow-1;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.component.scss:19]: We're removing this.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.module.ts
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.module.ts:1]: Make it a standalone component, and remove the module.

Comment (freddyucv) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.module.ts:1]: done https://github.com/dotCMS/core/pull/24807/commits/c88ed436a200dfcc0bf8860224bfca0a1d150355

File: core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.component.scss (hunk 1 above)
Comment (freddyucv) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.component.scss]: done https://github.com/dotCMS/core/pull/24807/commits/e2fe9bd3656d5d95428847c2b7389c64979457c3#diff-b9de41344c7150e006f9bb8d4465c2154bcb2c92f875a9f22013481667a335f7R18

File: core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.component.scss (hunk 2 above)
Comment (freddyucv) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-link/dot-link.component.scss]: done https://github.com/dotCMS/core/pull/24807/commits/e2fe9bd3656d5d95428847c2b7389c64979457c3#diff-b9de41344c7150e006f9bb8d4465c2154bcb2c92f875a9f22013481667a335f7L19

**Review by oidacra — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**